lib/kubeclient/common.rb

```diff
@@ -91,52 +138,60 @@ def build_namespace_prefix(namespace)
 
     public
 
-    def self.define_entity_methods(entity_types)
-      entity_types.each do |klass, entity_type|
-        entity_name = underscore_entity(entity_type)
-        entity_name_plural = pluralize_entity(entity_name)
+    def self.resource_class(class_owner, entity_type)
+      class_owner.const_get(entity_type, false)
+    rescue NameError
+      class_owner.const_set(
+        entity_type,
+        Class.new(RecursiveOpenStruct) do
+          def initialize(hash = nil, args = {})
+            args[:recurse_over_arrays] = true
+            super(hash, args)
+          end
+        end
+      )
+    end
 
+    def define_entity_methods
+      @entities.values.each do |entity|
```

Comment:
you mean with the list of verbs? each method has different parameters set so I'm not sure how that would work